### DISCUSSION TITLE
Representing quantities without an associated ingredient.

### DISCUSSION THREAD

**Discussion by Earnestly** [cooklang/spec#96] [Ideas]
Consider `separate the dough into 20g-25g pieces`. It would be nice for the quantity `20g-25g` to be represented in such a way that allows its units to be converted like those specified along with `@` ingredients.

**Comment by jphastings** [comment]
For syntax, I think omitting the ingredient would be simplest here: `separate the dough into %{20%g} pieces`. (I'm not sure how cooklang supports a range of values right now, perhaps `separate the dough into %{20%g}-%{25%g} pieces` is closest to what you're looking for?)

**Reply by dubadub** [reply]
Yes, can be as simple as `{20-25%g}` or nameless ingredient `@{20-25%g}` (for consistency)

**Comment by FranGarc** [comment]
But... what's the purpose of using markup for something that won't appear in the ingredients list? From what I can see on the android app, there's no specific representation of the quantities in the named ingredients. It's not like the timer notation that renders into a link that opens a timer.

**Comment by yutotakano** [comment]
I would like this for representing numbers that should scale with the user's input! For example, with bread dough, I would like to say "Now split the dough into %{2} sections." If the user scaled up the recipe by a factor of 2, it should show "split the dough into 4 sections", so the resulting section sizes are consistent.

**Reply by tmlmt** [reply]
I had also identified that need and have implemented it as an extension (see [docs](https://cooklang-parser.tmlmt.com/v3/guide-extensions.html#arbitrary-scalable-quantities)) in my (alternative) parser. You can see it in action in the [playground](https://cooklang-parser.tmlmt.com/playground/) and see if that was what you meant by it. This could serve as a proof of concept for potential implementation in the official cooklang specs.

**Reply by yutotakano** [reply]
Yes!!!!!!
